src/app/products/products.component.html

```diff
@@ -45,3 +45,9 @@ <h1></h1>
   <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
   <tr mat-row *matRowDef="let row; columns: displayedColumns"></tr>
 </table>
+
+<a routerLink="/create-product">
+  <button mat-button class="mat-button" color="custom">
+    Create new product
+  </button>
+</a>
```

Comment:
лучше всего все-таки будет так

```suggestion
<a mat-button class="mat-button" color="custom" routerLink="/create-product">
  Create new product
</a>
```